File: python/e1.py
"""
Find the sum of all the multiples of 3 or 5 below 1000. (fizzbuzz++)
"""

def Euler_1(n = 1000, divisors=(3,5)):
    #verified.
    return sum(i for i in range(0, n)
                if any(i % d == 0 for d in divisors))

def test_1():
    assert Euler_1() == 233168

if __name__ == "__main__":
    print(Euler_1())

File: ruby/e1.rb
# Find the sum of all the multiples of 3 or 5 below 1000 (fizzbuzz++)

def euler_1(n)
  return (0..n-1).select {|i| i % 3 == 0 || i % 5 == 0 }.sum
end

puts euler_1(1000)
